internal/controller/api/v1alpha1/installenv_types.go

```diff
@@ -22,6 +22,49 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+const (
+	ImageStateCreated        = "Image has been created"
+	ImageStateFailedToCreate = "Failed to create image"
+)
+
+// ClusterReference represents a Cluster Reference. It has enough information to retrieve cluster
+// in any namespace
+type ClusterReference struct {
+	// Name is unique within a namespace to reference a cluster resource.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Namespace defines the space within which the cluster name must be unique.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// InstallEnvConditionType is a valid value for InstallEnvCondition.Type
+type InstallEnvConditionType string
+
+const (
+	ImageProgressCondition InstallEnvConditionType = "ImageProgress"
```

Comment:
Not sure if this has been mentioned before, but I found these to be excellent resources when thinking about how to structure conditions.

* https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#conditions
* https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

So it may be better to make this ConditionType something like `ImageCreated`.

Reply:
Done